Source/FactoryGame/Private/FGResearchTree.cpp

```diff
@@ -33,38 +33,81 @@ TArray<class UFGAvailabilityDependency*> UFGResearchTree::GetUnlockDependencies(
+		}
+		return NodeDataStructProperty;
+	}
+
+	FClassProperty* GetSchematicStructProperty(FStructProperty* NodeDataStructProperty)
+	{
+		static FClassProperty* SchematicStructProperty = NULL;
+		if (NodeDataStructProperty && !SchematicStructProperty) {
+			// Generated structs names contain GUIDs, so we can't use FindPropertyByName
+			for (FField* StructProp = NodeDataStructProperty->Struct->ChildProperties; StructProp; StructProp = StructProp->Next) {
+				if (FClassProperty* ClassProp = CastField<FClassProperty>(StructProp)) {
+					if (ClassProp->MetaClass->IsChildOf(UFGSchematic::StaticClass())) {
+						SchematicStructProperty = ClassProp;
+						break;
+					}
+				}
+			}
+		}
+		return SchematicStructProperty;
+	}
+}
+
 EDataValidationResult UFGResearchTree::IsDataValid(TArray<FText>& ValidationErrors) {
 	// MODDING IMPLEMENTATION
 	EDataValidationResult ValidationResult = Super::IsDataValid(ValidationErrors);
 
+	// Exit early if the research tree has no nodes
+	const TArray<UFGResearchTreeNode*> Nodes = UFGResearchTree::GetNodes(GetClass());
+	if (Nodes.IsEmpty()) {
+		return ValidationResult;
+	}
+
 	// UFGResearchTreeNode::GetNodeSchematic is not implemented due to placeholder assets
 	// so we read its property directly
-	
-	static FStructProperty* NodeDataStructProperty = NULL;
-	static FClassProperty* SchematicStructProperty = NULL;
-	static UClass* ResearchTreeNodeClass = NULL;
-
-	// Lazily initialize research tree node reflection properties for faster access
-	if (ResearchTreeNodeClass == NULL) {
-		ResearchTreeNodeClass = LoadClass<UFGResearchTreeNode>(NULL, TEXT("/Game/FactoryGame/Schematics/Research/BPD_ResearchTreeNode.BPD_ResearchTreeNode_C"));
-		check(ResearchTreeNodeClass);
-		//Make sure class is not garbage collected
-		ResearchTreeNodeClass->AddToRoot();
+	UClass* ResearchTreeNodeClass = GetResearchTreeNodeClass();
+	FStructProperty* NodeDataStructProperty = GetNodeDataStructProperty(ResearchTreeNodeClass);
+	FClassProperty* SchematicStructProperty = GetSchematicStructProperty(NodeDataStructProperty);
 
-		NodeDataStructProperty = CastFieldChecked<FStructProperty>(ResearchTreeNodeClass->FindPropertyByName(TEXT("mNodeDataStruct")));
-
-		// Generated structs names contain GUIDs, so we can't use FindPropertyByName
-		for(FField* StructProp = NodeDataStructProperty->Struct->ChildProperties; StructProp; StructProp = StructProp->Next) {
-			if (FClassProperty* ClassProp = CastField<FClassProperty>(StructProp)) {
-				if (ClassProp->MetaClass->IsChildOf(UFGSchematic::StaticClass())) {
-					SchematicStructProperty = ClassProp;
-					break;
-				}
-			}
-		}
+	if (!ResearchTreeNodeClass || !NodeDataStructProperty || !SchematicStructProperty) {
+		ValidationErrors.Add(NSLOCTEXT("ResearchTree", "Validation_ReflectionFailed", "Could not retrieve properties to validate using reflection, validation not performed"));
```

Comment:
This is still an error. No idea if it will cause problems with the Asset Generator, though.

Reply:
I don't think it would cause any issues, since the asset generator still saves the changes, and nothing should depend on data in the CDO of a research tree.

But the asset generator should also provide the SkipDefaultObjectValidation flag (or something like that) to the blueprint compilation, so that the blueprint is still compiled after every change